juju/juju
# Adds InterfaceInfos type to core/network

## Description of change

This patch adds a new `InterfaceInfos` type to core network for operating with collections of `InterfaceInfo`.

Included is a utility method to iterate over the collection in order of device hierarchy, so we can process each interface with parents always before children.

Unused sort functionality is removed and tests specific to `InterfaceInfo` have been relocated from the old `network` package to `core/network`.

## QA steps

The new type and its utility are not yet recruited; unit tests verify logic additions.

## Documentation changes

None.

## Bug reference

N/A
